/**
 * Copyright (c) 2014-present, The osquery authors
 *
 * This source code is licensed as defined by the LICENSE file found in the
 * root directory of this source tree.
 *
 * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
 */

syntax = "proto3";

package containerd.linux_runtime.runc;

message RuncOptions {
	string runtime = 1;
	string runtime_root = 2;
	string criu_path = 3;
	bool systemd_cgroup = 4;
}

message CreateOptions {
	bool no_pivot_root = 1;
	bool open_tcp = 2;
	bool external_unix_sockets = 3;
	bool terminal = 4;
	bool file_locks = 5;
	repeated string empty_namespaces = 6;
	string cgroups_mode = 7;
	bool no_new_keyring = 8;
	string shim_cgroup = 9;
	uint32 io_uid = 10;
	uint32 io_gid = 11;
	string criu_work_path = 12;
	string criu_image_path = 13;
}

message CheckpointOptions {
	bool exit = 1;
	bool open_tcp = 2;
	bool external_unix_sockets = 3;
	bool terminal = 4;
	bool file_locks = 5;
	repeated string empty_namespaces = 6;
	string cgroups_mode = 7;
	string work_path = 8;
	string image_path = 9;
}

message ProcessDetails {
	string exec_id = 1;
}